Bottom: 90123bf005a1be20510d273550a93d87ce7bd12f
Top:    340a70c785b621cbfa1d4fbf959ded593a13364d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-21 02:33:49 +0000

Add profiling to file load

















---

diff --git a/src/file.c b/src/file.c
index b6c67b9..018b272 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
